src/nomad_material_processing/__init__.py

```diff
@@ -49,287 +52,364 @@
-        ''',
+        """,
     )
 
 
-class Substrate(CompositeSystem):
-    '''
+class SubstrateSlab(CompositeSystem):
+    """
     A thin free standing sheet of material. Not to be confused with the substrate role
     during a deposition, which can be a `Substrate` with `ThinFilm`(s) on it.
-    '''
+    """
+
     m_def = Section()
     supplier = Quantity(
         type=str,
-        description='The supplier of the current substrate specimen.',
+        description="The supplier of the current substrate specimen.",
         a_eln=ELNAnnotation(
             component=ELNComponentEnum.StringEditQuantity,
-            label='Name of Supplier',
-        )
+            label="Name of Supplier",
+        ),
     )
     supplier_id = Quantity(
         type=str,
-        description='An ID string that is unique from the supplier.',
+        description="An ID string that is unique from the supplier.",
         a_eln=ELNAnnotation(
             component=ELNComponentEnum.StringEditQuantity,
-            label='Supplier ID',
+            label="Supplier ID",
         ),
     )
     lab_id = Quantity(
         type=str,
         a_eln=ELNAnnotation(
             component=ELNComponentEnum.StringEditQuantity,
-            label='Substrate ID',
+            label="Substrate ID",
         ),
     )
 
 
-class CrystallineSubstrate(Substrate):
-    '''
+class CrystallineSubstrate(SubstrateSlab):
+    """
     The substrate defined in this class is composed of periodic arrangement of atoms
     and shows typical features of a crystal structure.
-    '''
+    """
+
     m_def = Section()
     geometry = SubSection(
         section_def=Geometry,
-        description='Section containing the geometry of the substrate.',
+        description="Section containing the geometry of the substrate.",
     )
     crystal_properties = SubSection(
         section_def=SubstrateCrystalProperties,
-        description='Section containing the crystal properties of the substrate.',
+        description="Section containing the crystal properties of the substrate.",
     )
     dopants = SubSection(
         section_def=Dopant,
         repeats=True,
-        description='''
+        description="""
         Repeating section containing information on any dopants in the substrate.
-        ''',
+        """,
     )
 
 
 class ThinFilm(CompositeSystem):
-    '''
+    """
     A thin film of material which exists as part of a stack.
-    '''
+    """
+
     m_def = Section()
 
     geometry = SubSection(
         section_def=Geometry,
-        description='Section containing the geometry of the thin film.',
+        description="Section containing the geometry of the thin film.",
+    )
+
+
+class ThinFilmReference(CompositeSystemReference):
+    """
+    Class autogenerated from yaml schema.
+    """
+
+    m_def = Section()
+    lab_id = Quantity(
+        type=str,
+        a_eln=ELNAnnotation(
+            component="StringEditQuantity",
+        ),
+    )
```

Comment:
Is there a reason for overloading the `lab_id` attribute from `EntityReference` here?

Reply:
there is not :)